synapse/handlers/oidc_handler.py

```diff
@@ -282,6 +282,13 @@ def __init__(
         self._config = provider
         self._callback_url = hs.config.oidc_callback_url  # type: str
 
+        # TODO: This is pretty much a hack to get the path specified by public_baseurl.
+        # It'd probably be nicer to have a config option that lets you specify a custom
+        # path, which we'd then use here.
```

Comment:
why would we want a custom path here?

Reply:
I meant that instead of specifying one's `public_baseurl` as, say, `https://example.com/specific/deployment`, you would have one option for the hostname, and another for the path. This would save us from needing to parse the path from the URL, though I suppose that isn't probably wouldn't buy us *that* much.

Reply:
yeah, I think that would be more confusing than helpful.

Reply:
... on which basis, can you update the TODO?